### PR TITLE
Implemented use statement for baseClass

### PR DESCRIPTION
**Pull request by geolysis**
Base classes in a different namespace are currently not supported. It would be great, if you would consider this change, that this becomes a part of the standard functionality.

Thanks, Andreas

### CODE REVIEW

File: src/Propel/Generator/Builder/Om/ObjectBuilder.php [hunk 1]
```diff
@@ -282,6 +282,14 @@ protected function addClassBody(&$script)
             '\Propel\Runtime\Map\TableMap'
         );
 
+        $baseClass = $this->getBaseClass();
+        if (strrpos($baseClass, '\\') !== FALSE) {
+            $classesToDeclare[] = $baseClass;
+        }
+
+        $reflectionMethod = new \ReflectionMethod($this, 'declareClasses');
```
Comment (lbarulski) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: why not: 
```php
call_user_func_array([$this, 'declareClasses'], $classesToDeclare);
```
?

Comment (geolysis) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: This comes down to the preferred programming style. My preference is an object orientated approach. Of course `call_user_func_array` is valid solution, as well.

Comment (staabm) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: I would guess that the RefecltionMethod is much slower then cufa

Comment (marcj) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: Well, I'd completely not use any of those stuff.

```php
$baseClass = $this->getBaseClass();
if (false !=== strrpos($baseClass, '\\')) {
    $this->declareClasses($baseClass);
}
```

Doing it this way is easier to read, less modification and no user_func-call

**Comment by geolysis** [comment]
Perfect solution, Marc.

**Comment by geolysis** [comment]
Happy to implement it now with the adjustment made?

**Comment by marcj** [comment]
Well, thanks @geolysis :)

**Pull request merged**